### ISSUE TITLE
Warn on semvar incompatabilities

### ISSUE THREAD

**Issue by Deedasmi** [George3d6/Inquisitor#38]
The agent and receptor should provide compilation warnings when plugins have a lower 'major release' version than the agent/receptor itself. i.e. Agent version 1.0.0, Alive plugin version 0.9.5.

This should also apply to agent_lib, shared_lib, etc.

**Comment by George3d6** [comment]
How would this be even done ? Maybe implemented as part of the plugin creator macro using the metadata from the cargo files... I guess, seems a bit complex, maybe worth looking into but I don't think it would come in that handy and it would add even more complexity to the plugin initialization macro.

Overall, if the plugins are incompatible with the API the code just won't compile, if they are, they should still work more or less as intended.

**Comment by Deedasmi** [comment]
My thought was build.rs has access to the cargo metadata and can glance at versions. It should be able to panic the build as soon as it finds an issue. Won't change plugin initialization at all.

Reasoning behind it is because if there is an API issue the error messages don't always say what plugin has the API issue. If you recall the output I posted a couple weeks ago, one API error caused every plugin to throw compilation errors. This is just an easy way to make the 1.0 transition easier.

Although we could make it work for all API changes if we also checked minor version on major version 0 with your semvar recommendation....

Richard Petrie
Kansas State University

Sent from a mobile device

On Apr 12, 2018 10:50 AM, George <notifications@github.com> wrote:

How would this be even done ? Maybe implemented as part of the plugin creator macro using the metadata from the cargo files... I guess, seems a bit complex, maybe worth looking into but I don't think it would come in that handy and it would add even more complexity to the plugin initialization macro.

Overall, if the plugins are incompatible with the API the code just won't compile, if they are, they should still work more or less as intended.
